services/datalad/datalad_service/tasks/publish.py

```diff
@@ -145,6 +149,43 @@ def publish_github_async(store, dataset, snapshot, github_remote):
     ds = store.get_dataset(dataset)
     publish_target(ds, github_remote, snapshot)
 
+def delete_s3_sibling(dataset_id, siblings, realm):
+    sibling = get_sibling_by_name(realm.s3_remote, siblings)
+    if sibling:
+        try:
+            session = boto3.Session(
+                aws_access_key_id=AWS_ACCESS_KEY_ID,
+                aws_secret_access_key=AWS_SECRET_ACCESS_KEY,
+            )
+            s3 = session.resource('s3')
+            bucket = s3.Bucket(realm.s3_bucket)
+            bucket.objects.filter(Prefix=f'{dataset_id}/').delete()
```

Comment:
It looks like this does not delete the versions, only adds a delete marker. That wouldn't make old versions unavailable, I think this delete step needs to be replaced with one which crawls all versions of objects within the prefix and issues a delete for each version.